dashboard/streamlit_app.py

```diff
@@ -1,73 +1,177 @@
+    """
+
+    with conn.cursor() as cur:
+        cur.execute(query)
+        data = cur.fetchall()
+
+    data_df = pd.DataFrame(data, columns=CSV_COLUMNS)
+    data_df["last_watered"] = pd.to_datetime(data_df["last_watered"])
+    data_df["recording_taken"] = pd.to_datetime(data_df["recording_taken"])
+
+    return data_df
+
+
+def get_selected_archive() -> pd.DataFrame:
+    """ Creates the dropdown and which archived file 
+    to select and return for display."""
+    with st.sidebar:
+        st.sidebar.title("Dropdown")
+
+        csv_files = [f for f in listdir(
+            "archived_data") if f.endswith('.csv')]
+
+        # Dropdown to select a CSV file.
+        selected_csv = st.selectbox(
+            "Select an archived file.", csv_files, on_change=on_toggle_or_archive_change)
+        data = pd.read_csv(path.join("archived_data", selected_csv))
+    return data
 
-def get_live_data(db_conn: connection) -> pd.DataFrame:
-    """Function that gets the required data from the database connection 
-    and puts all data into pandas DataFrame form"""
-    cursor = db_conn.cursor()
-    cursor.execute("SELECT * FROM plant")
-    names = [x[0] for x in cursor.description]
-    live_plants_data = pd.DataFrame(cursor.fetchall(), columns=names)
-    return live_plants_data
 
+def on_toggle_or_archive_change() -> None:
+    """
+    Sets the index back to 0 for all of the graphs upon changing
+    the toggle status or which archive is being viewed
+    """
+    st.session_state.start_index = 0
 
-def get_archived_data() -> pd.DataFrame:
-    """Function that gets the required data from the s3 bucket and
-    puts all data into pandas DataFrame form"""
 
-    # Download all CSV files from the archive s3 bucket
+def switch_data(db_connection: connection) -> pd.DataFrame:
+    """
+    Returns either the live data or the archived data 
+    based on if toggle is on. (The default is live data)
+    """
+    data = get_live_database(db_connection)
+    toggle_on = st.sidebar.toggle(
+        label="Toggle for Archived Data", value=False, on_change=on_toggle_or_archive_change)
+    if toggle_on:
+        data = get_selected_archive()
+    return data
 
-    csv_files = [f for f in os.listdir("archived_data") if f.endswith('.csv')]
-    dfs = []
 
-    for csv in csv_files:
-        csv_to_df = pd.read_csv("archived_data/"+csv)
-        dfs.append(csv_to_df)
+def dashboard_header() -> None:
+    """Creates a header for the dashboard."""
+    st.title("ONTHEB-RINK-OFEXTINCTION")
+    st.markdown("An app for Liverpool Natural History Museum,\
+                        #monitoring the health of our plants")
 
-    archived_plants_data = pd.concat(dfs, ignore_index=True)
 
-    # Delete all CSV files from the archived_data folder locally
+def plot_temp_for_plants(data_df) -> None:
+    """
+    creates a bar chart with temperature in y axis
+    and the plant name in x axis.
+    """
```

Comment:
Multi-line docstrings should start with text on the same line as the start of the `"""` and end with `"""` by itself as per pep8 requirements. https://peps.python.org/pep-0008/#documentation-strings
Additionally, comments should follow sentence case. https://peps.python.org/pep-0008/#comments